src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Withdraw.tsx

```diff
@@ -373,180 +386,129 @@ export const Withdraw: React.FC<WithdrawProps> = ({ accountId, onNext }) => {
-        const minLimitCryptoPrecision = bn(outboundFeeCryptoBaseUnit).div(
-          bn(10).pow(asset.precision),
-        )
-        const minLimit = `${minLimitCryptoPrecision} ${asset.symbol}`
-        return translate('trade.errors.amountTooSmall', {
-          minLimit,
-        })
+        if (valueCryptoPrecision.isEqualTo(0)) return ''
+        return hasValidBalance || 'common.insufficientFunds'
+      } catch (e) {
+        // This should never happen since all errors are monadic, but allows us to use the finally block
+        console.error(e)
+      } finally {
+        setQuoteLoading(false)
+        dispatch({ type: ThorchainSaversWithdrawActionType.SET_LOADING, payload: false })
       }
-
-      const hasValidBalance = fiat.gt(0) && valueCryptoPrecision.gt(0) && fiat.gte(value)
-      if (valueCryptoPrecision.isEqualTo(0)) return ''
-      return hasValidBalance || 'common.insufficientFunds'
     },
     [
-      amountAvailableCryptoPrecision,
-      asset.precision,
-      asset.symbol,
-      assetMarketData.price,
-      maybeOutboundFeeCryptoBaseUnit,
-      maybeWithdrawGasEstimateCryptoBaseUnit,
+      opportunityData,
+      accountId,
+      asset,
+      getOutboundFeeCryptoBaseUnit,
       translate,
+      getWithdrawGasEstimateCryptoBaseUnit,
+      amountAvailableCryptoPrecision,
+      dispatch,
     ],
   )
 
-  useEffect(() => {
-    if (!(accountId && inputValues && asset && opportunityData?.stakedAmountCryptoBaseUnit)) return
-    const { cryptoAmount } = inputValues
-    const amountCryptoBaseUnit = toBaseUnit(cryptoAmount, asset.precision)
-
-    if (bn(amountCryptoBaseUnit).isZero()) return
-
-    const debounced = debounce(async () => {
-      setQuoteLoading(true)
+  const validateFiatAmount = useCallback(
+    (value: string) => {
+      if (!(opportunityData && accountId && dispatch)) return false
+      dispatch({ type: ThorchainSaversWithdrawActionType.SET_LOADING, payload: true })
 
       try {
-        const withdrawBps = getWithdrawBps({
-          withdrawAmountCryptoBaseUnit: amountCryptoBaseUnit,
-          stakedAmountCryptoBaseUnit: opportunityData.stakedAmountCryptoBaseUnit ?? '0',
-          rewardsAmountCryptoBaseUnit: opportunityData.rewardsCryptoBaseUnit?.amounts[0] ?? '0',
-        })
-
-        const _quote = await getThorchainSaversWithdrawQuote({ asset, accountId, bps: withdrawBps })
-        const { dust_amount, slippage_bps } = _quote
-        const percentage = bnOrZero(slippage_bps).div(BASE_BPS_POINTS).times(100)
-
-        // total downside (slippage going into position) - 0.007 ETH for 5 ETH deposit
-        const cryptoSlippageAmountPrecision = bnOrZero(cryptoAmount).times(percentage).div(100)
-        setSlippageCryptoAmountPrecision(cryptoSlippageAmountPrecision.toString())
-
-        const _dustAmountCryptoBaseUnit = toBaseUnit(fromThorBaseUnit(dust_amount), asset.precision)
+        const crypto = bnOrZero(amountAvailableCryptoPrecision.toPrecision())
```

Comment:
This just moved, not an added line. Most likely no point to do the `toPrecision()` dance here, but not going to introduce more functional changes than necessary